src/renderer/base/RenderClusterIterator.cpp

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+#include "../inc/RenderClusterIterator.hpp"
+
+#include "../../buffer/out/textBufferCellIterator.hpp"
+
+using namespace Microsoft::Console::Render;
+
+RenderClusterIterator::RenderClusterIterator(TextBufferCellIterator& cellIter) :
+    _cellIter(cellIter),
+    _cluster(L"", 0),
+    _attr((*cellIter).TextAttr()),
+    _distance(0),
+    _exceeded(false)
+{
+    _GenerateCluster();
+}
+
+RenderClusterIterator::operator bool() const noexcept
+{
+    return !_exceeded;
+}
+
+// Routine Description:
+// - Compares two iterators to see if they're pointing to the same position in the same buffer
+// Arguments:
+// - it - The other iterator to compare to this one.
+// Return Value:
+// - True if it's the same text buffer and same cell position. False otherwise.
+bool RenderClusterIterator::operator==(const RenderClusterIterator& it) const
+{
+    return _attr == it._attr &&
+           &_cellIter == &it._cellIter &&
+           &_distance == &it._distance &&
+           _exceeded == it._exceeded;
+}
+
+// Routine Description:
+// - Compares two iterators to see if they're pointing to the different positions in the same buffer or different buffers entirely.
+// Arguments:
+// - it - The other iterator to compare to this one.
+// Return Value:
+// - True if it's the same text buffer and different cell position or if they're different buffers. False otherwise.
+bool RenderClusterIterator::operator!=(const RenderClusterIterator& it) const
+{
+    return !(*this == it);
+}
+
+RenderClusterIterator& RenderClusterIterator::operator+=(const ptrdiff_t& movement)
+{
+    ptrdiff_t move = movement;
+    auto newCellIter = _cellIter;
+    size_t cols = 0;
+    while (move > 0 && !_exceeded)
+    {
+        if (!(*newCellIter).DbcsAttr().IsTrailing())

```

Comment:
This is an attempt to let `RenderClusterIterator` handle `dbcs` column, which I think is useless now. Pretty hacky to me.